### PR TITLE
test: Replace gocheck with built-in go test (part 1)

### PR DESCRIPTION
**Pull request by sayboras**
### Description

Please refer to individual commit for more details. Just a note this is just a first cut, there are still some other packages pending to update.

Relates: https://github.com/cilium/cilium/issues/28596

### CODE REVIEW

**Review by christarazi — APPROVED** [review]
LGTM for alibabacloud changes

**Review by aspsk — APPROVED** [review]

**Comment by sayboras** [comment]
Build commit failure is due to time as we are having quite a number of small commits. It  has been verified locally as per below

```
$ time git rebase --exec "make build -j $(nproc)" main
...
Successfully rebased and updated refs/heads/pr/tammach/replace-gocheck.
git rebase --exec "make build -j $(nproc)" main  2119.71s user 571.66s system 1410% cpu 3:10.86 total
```

**Review by gandro — APPROVED** [review]
Amazing, thank you!

**Review by nebril — APPROVED** [review]
Amazing! I don't look forward to backporting unit test changes though ;_;

File: pkg/bpf/map_linux_test.go [hunk 1]
```diff
@@ -51,64 +43,55 @@ func (k *TestKey) New() MapKey    { return &TestKey{} }
 func (v *TestValue) String() string { return fmt.Sprintf("value=%d", v.Value) }
 func (v *TestValue) New() MapValue  { return &TestValue{} }
 
-var _ = Suite(&BPFPrivilegedTestSuite{})
-
-func (s *BPFPrivilegedTestSuite) SetUpSuite(c *C) {
-	testutils.PrivilegedTest(c)
+func setup(tb testing.TB) {
```
Comment (ti-mo) [inline pkg/bpf/map_linux_test.go:46]: Make this return a `*bpf.Map` so the global `testMap` can go.

File: pkg/bpf/map_linux_test.go [hunk 2]
```diff
@@ -51,64 +43,55 @@ func (k *TestKey) New() MapKey    { return &TestKey{} }
...
 	maxEntries = 16
-
-	testMap = NewMap("cilium_test",
-		ebpf.Hash,
-		&TestKey{},
-		&TestValue{},
-		maxEntries,
-		BPF_F_NO_PREALLOC,
-	).WithCache()
+	testMap    *Map
```
Comment (ti-mo) [inline pkg/bpf/map_linux_test.go:79]: See my comments above, we should remove this one. :slightly_smiling_face:

File: pkg/bpf/map_linux_test.go [hunk 3]
```diff
@@ -424,100 +415,104 @@ func (s *BPFPrivilegedTestSuite) TestSubscribe(c *C) {
...
+	require.Equal(t, map[string][]string{
 		"key=0":   {"value=0"},
 		"key=105": {"value=205"},
 		"key=106": {"value=206"},
-	})
+	}, dump5)
 }
 
-// TestDumpReliablyWithCallbackOveralapping attempts to test that DumpReliablyWithCallback
+// TestDumpReliablyWithCallbackOverlapping attempts to test that DumpReliablyWithCallback
```
Comment (ti-mo) [inline pkg/bpf/map_linux_test.go:507]: :ok_hand: :ok_hand: :ok_hand:

File: pkg/bpf/map_linux_test.go [hunk 4]
```diff
@@ -51,64 +43,55 @@ func (k *TestKey) New() MapKey    { return &TestKey{} }
...
 
+	testMap = NewMap("cilium_test",
+		ebpf.Hash,
+		&TestKey{},
+		&TestValue{},
+		maxEntries,
+		BPF_F_NO_PREALLOC,
+	).WithCache()
+
 	if err := testMap.OpenOrCreate(); err != nil {
```
Comment (ti-mo) [inline pkg/bpf/map_linux_test.go:63]: No real need to pin this map afaik, let's change it:
```suggestion
	if err := testMap.CreateUnpinned(); err != nil {
```

This also allows us to get rid of the `os.Remove()` in tb.Cleanup.

File: pkg/cidr/cidr_test.go [hunk 5]
```diff
@@ -183,7 +172,9 @@ func (t *CidrTestSuite) TestEqual(c *check.C) {
 		},
 	}
 	for _, tt := range tests {
-		c.Assert(tt.fields.n.Equal(tt.args.o), check.Equals, tt.want, check.Commentf("Test Name: %s", tt.name))
+		t.Run(tt.name, func(t *testing.T) {
```
Comment (ti-mo) [inline pkg/cidr/cidr_test.go:175]: Not sure if I'd create subtests here, their names are completely meaningless. You're already passing tt.name as the last argument to `require.Equal`, so it's already clear which assertion failed.

**Review by ti-mo — APPROVED** [review]
Thank you so much for picking this up! :pray:

File: bugtool/cmd/helper_test.go [hunk 6]
```diff
@@ -51,74 +43,63 @@ func (l *logWrapper) Write(p []byte) (n int, err error) {
...
-}
-
 // TestWalkPath tests that with various different error types, we can safely
 // back out and continue with the filepath walk. This allows gathering of other
 // information in a bugtool run when there's an issue with a particular file.
-func (b *BugtoolSuite) TestWalkPath(c *C) {
-	w := newWalker(baseDir, tmpDir, &dummyTarWriter{}, &logWrapper{c.Logf})
-	c.Assert(w, NotNil)
+func TestWalkPath(t *testing.T) {
+	baseDir, tmpDir = t.TempDir(), t.TempDir()
```
Comment (tklauser) [inline bugtool/cmd/helper_test.go:50]: nit: could use `:=` here and drop the `baseDir` and `tmpDir` global vars.

File: cilium-dbg/cmd/bpf_ipcache_get_test.go [hunk 7]
```diff
@@ -71,12 +66,11 @@ func (s *BPFIPCacheGetSuite) TestGetLPMValue(c *C) {
...
 	for _, tt := range tests {
 		v, exists := getLPMValue(mustParseIP(tt.ip), entries)
-
-		c.Assert(exists, Equals, tt.hasIdentity, Commentf("No identity was found for ip '%s': wanted '%s'", tt.ip, tt.identity))
+		require.Equal(t, exists, tt.hasIdentity, fmt.Sprintf("No identity was found for ip '%s': wanted '%s'", tt.ip, tt.identity))
 
 		if exists {
 			identity := v.([]string)
-			c.Assert(identity, checker.DeepEquals, tt.identity, Commentf("Wrong identity was retrieved for ip %s", tt.ip))
+			require.Equal(t, len(identity), len(tt.identity), fmt.Sprintf("Wrong number of identities was retrieved for ip %s", tt.ip))
```
Comment (tklauser) [inline cilium-dbg/cmd/bpf_ipcache_get_test.go:73]: Shouldn't we compare the slices' content here as well?

File: cilium-dbg/cmd/bpf_metrics_list_test.go [hunk 8]
```diff
@@ -98,19 +94,18 @@ func (s *BPFMetricsMapSuite) TestDumpMetrics(c *C) {
...
 	err := json.Unmarshal([]byte(rawDump), &got)
-	c.Assert(err, IsNil, Commentf("invalid JSON output: '%s', '%s'", err, rawDump))
+	require.NoError(t, err, "invalid JSON output: '%s', '%s'", err, rawDump)
 
 	sort.Slice(got, func(i, j int) bool {
 		return got[i].Packets <= got[j].Packets
 	})
-
-	c.Assert(want, checker.DeepEquals, got)
+	require.Equal(t, len(want), len(got))
```
Comment (tklauser) [inline cilium-dbg/cmd/bpf_metrics_list_test.go:110]: Shouldn't we compare the slices' content here as well?

File: daemon/cmd/status_test.go [hunk 9]
```diff
@@ -37,18 +32,18 @@ var (
 	}
 )
 
-func (g *GetNodesSuite) SetUpTest(c *C) {
+func setup(t *testing.T) {
+	var err error
```
Comment (tklauser) [inline daemon/cmd/status_test.go:36]: Add `t.Helper()` call here to mark it as a test helper functions and so failures will be attributed to the caller.

```suggestion
func setup(t *testing.T) {
	t.Helper()
	
	var err error
```

File: pkg/alibabacloud/eni/node_test.go [hunk 10]
```diff
@@ -29,7 +30,7 @@ var (
 	metricsapi = metricsmock.NewMockMetrics()
 )
 
-func (e *ENISuite) SetUpTest(c *check.C) {
+func setup(tb testing.TB) {
 	limits.Update(map[string]ipamTypes.Limits{
```
Comment (tklauser) [inline pkg/alibabacloud/eni/node_test.go:34]:
```suggestion
func setup(tb testing.TB) {
	tb.Helper()

	limits.Update(map[string]ipamTypes.Limits{
```

File: pkg/aws/eni/eni_gc_test.go [hunk 11]
```diff
@@ -5,17 +5,18 @@ package eni
...
+	"github.com/stretchr/testify/require"
 
 	ec2mock "github.com/cilium/cilium/pkg/aws/ec2/mock"
 	"github.com/cilium/cilium/pkg/controller"
 	"github.com/cilium/cilium/pkg/inctimer"
 	"github.com/cilium/cilium/pkg/logging"
 )
 
-func waitForControllerRun(c *check.C, controller *controller.Manager, name string, expectedCount int64) {
+func waitForControllerRun(t *testing.T, controller *controller.Manager, name string, expectedCount int64) {
```
Comment (tklauser) [inline pkg/aws/eni/eni_gc_test.go:19]: Add `t.Helper` here while at it:

```suggestion
func waitForControllerRun(t *testing.T, controller *controller.Manager, name string, expectedCount int64) {
	t.Helper()

```

File: pkg/aws/eni/node_manager_test.go [hunk 12]
```diff
@@ -54,58 +53,62 @@ var (
 	metricsapi = metricsmock.NewMockMetrics()
 )
 
-func (e *ENISuite) SetUpTest(c *check.C) {
+func setup(tb testing.TB) {
```
Comment (tklauser) [inline pkg/aws/eni/node_manager_test.go:56]:
```suggestion
func setup(tb testing.TB) {
	tb.Helper()

```

File: pkg/cgroups/manager/manager_test.go [hunk 13]
```diff
@@ -135,7 +125,7 @@ func newCgroupManagerTest(pMock providerMock, cg cgroup) *CgroupManager {
 	return initManager(pMock, cg, 0)
 }
 
-func (m *ManagerSuite) SetUpTest(c *C) {
+func setup(tb testing.TB) {
```
Comment (tklauser) [inline pkg/cgroups/manager/manager_test.go:128]:
```suggestion
func setup(tb testing.TB) {
	tb.Helper()

```

File: pkg/clustermesh/services_test.go [hunk 14]
```diff
@@ -55,18 +50,17 @@ type ClusterMeshServicesTestSuite struct {
...
 }
 
-var _ = Suite(&ClusterMeshServicesTestSuite{})
+var s *ClusterMeshServicesTestSuite
 
-func (s *ClusterMeshServicesTestSuite) SetUpSuite(c *C) {
-	testutils.IntegrationTest(c)
-}
+func setup(tb testing.TB) {
+	testutils.IntegrationTest(tb)
```
Comment (tklauser) [inline pkg/clustermesh/services_test.go:56]:
```suggestion
func setup(tb testing.TB) {
	tb.Helper()

	testutils.IntegrationTest(tb)
```

**Review by tklauser — CHANGES_REQUESTED** [review]

**Comment by joestringer** [comment]
Would it make sense to break this PR up and re-post multiple PRs by codeowner areas? Might make each step go faster. Otherwise this is a massive PR with ~30 commits and 4K LoC change.

**Comment by sayboras** [comment]
> Would it make sense to break this PR up and re-post multiple PRs by codeowner areas? Might make each step go faster. Otherwise this is a massive PR with ~30 commits and 4K LoC change.

Ah right, it's one of the reasons that I didn't send the patch earlier, as I am not sure how to make it easier for reviewer. 

Will break this PR into smaller one via codeower,  and incorporate all feedback received so far.

File: bugtool/cmd/helper_test.go (hunk 6 above)
Comment (sayboras) [reply bugtool/cmd/helper_test.go:50]: thanks, this is addressed in https://github.com/cilium/cilium/pull/32210

File: cilium-dbg/cmd/bpf_ipcache_get_test.go (hunk 7 above)
Comment (sayboras) [reply cilium-dbg/cmd/bpf_ipcache_get_test.go:73]: thanks, it's addressed in https://github.com/cilium/cilium/pull/32210

File: cilium-dbg/cmd/bpf_metrics_list_test.go (hunk 8 above)
Comment (sayboras) [reply cilium-dbg/cmd/bpf_metrics_list_test.go:110]: thanks, it's addressed in https://github.com/cilium/cilium/pull/32210

File: pkg/bpf/map_linux_test.go (hunk 1 above)
Comment (sayboras) [reply pkg/bpf/map_linux_test.go:46]: Thanks, this is addressed in https://github.com/cilium/cilium/pull/32220

File: pkg/bpf/map_linux_test.go (hunk 2 above)
Comment (sayboras) [reply pkg/bpf/map_linux_test.go:79]: thanks, this should be addressed in https://github.com/cilium/cilium/pull/32220

File: pkg/bpf/map_linux_test.go (hunk 4 above)
Comment (sayboras) [reply pkg/bpf/map_linux_test.go:63]: Thanks, this should be incorporated in https://github.com/cilium/cilium/pull/32220

File: pkg/cidr/cidr_test.go (hunk 5 above)
Comment (sayboras) [reply pkg/cidr/cidr_test.go:175]: As we are changing to require for assertion, this is to make sure that all the tests are run. Not to mention that we can have a capability to run one individual test.

**Pull request closed without merging**